snuba_sdk/formula.py

```diff
@@ -43,12 +43,35 @@ def validate(self) -> None:
                 raise InvalidFormulaError(
                     f"parameters of formula {self.operator.value} must be a Sequence"
                 )
+
+            # - Must have the same entities, and must have an entity (no formulas on just literals)
+            # - Must have the same groupbys
+            entity = None
+            groupby = None
             for param in self.parameters:
-                if not isinstance(param, (Formula, Timeseries, float, int)):
+                if not isinstance(param, (Timeseries, float, int)):
                     raise InvalidFormulaError(
                         f"parameter '{param}' of formula {self.operator.value} is an invalid type"
                     )
 
+                if isinstance(param, Timeseries):
+                    if entity is None:
+                        entity = param.metric.entity
+                    elif entity != param.metric.entity:
+                        raise InvalidFormulaError(
+                            "Formulas can only operate on a single entity"
+                        )
+
+                    if groupby is None:
+                        groupby = param.groupby
+                    elif groupby != param.groupby:
```

Comment:
Can we check for the equality of sets here in case they contain the same elements but out of order?